boards/arm/wio_terminal/wio_terminal.dts

```diff
@@ -0,0 +1,266 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+#include <atmel/samd5xx19.dtsi>
+#include "wio_terminal-pinctrl.dtsi"
+#include "grove_connectors.dtsi"
+#include "raspberrypi_40pins_connector.dtsi"
+#include <zephyr/dt-bindings/display/ili9xxx.h>
+
+/ {
+	model = "Wio Terminal";
+	compatible = "seeed,wio-terminal";
+
+	chosen {
+		zephyr,sram = &sram0;
+		zephyr,flash = &flash0;
+		zephyr,code-partition = &code_partition;
+		zephyr,display = &ili9341;
+	};
+
+	/* These aliases are provided for compatibility with samples */
+	aliases {
+		led0 = &led0;
+		sw0 = &user_button_0;
+		sw1 = &user_button_1;
+		sw2 = &user_button_2;
+		i2c0 = &sercom4;
+		i2c1 = &sercom3;
+		spi0 = &sercom5;
+		uart0 = &sercom2;
+		accel0 = &lis3dh;
+		sdhc0 = &sdhc0;
+	};
+
+	/* LEDs */
+	leds {
+		compatible = "gpio-leds";
+		led0: led_0 {
+			label = "LED";
+			gpios = <&porta 15 GPIO_ACTIVE_HIGH>;
+		};
+	};
+
+	/* Buttons */
+	gpio_keys {
+		compatible = "gpio-keys";
+		user_button_0: button_0 {
+			label = "User Button 0";
+			gpios = <&portc 26 GPIO_ACTIVE_LOW>;
+		};
+		user_button_1: button_1 {
+			label = "User Button 1";
+			gpios = <&portc 27 GPIO_ACTIVE_LOW>;
+		};
+		user_button_2: button_2 {
+			label = "User Button 2";
+			gpios = <&portc 28 GPIO_ACTIVE_LOW>;
+		};
+		joy_sel: joystick_selection {
+			label = "joystick selection";
+			gpios = <&portd 10 GPIO_ACTIVE_LOW>;
+		};
+		joy_down: joystick_down {
+			label = "joystick down";
+			gpios = <&portd 8 GPIO_ACTIVE_LOW>;
+		};
+		joy_up: joystick_up {
+			label = "joystick up";
+			gpios = <&portd 20 GPIO_ACTIVE_LOW>;
+		};
+		joy_left: joystick_left {
+			label = "joystick left";
+			gpios = <&portd 12 GPIO_ACTIVE_LOW>;
+		};
+		joy_right: joystick_right {
+			label = "joystick right";
+			gpios = <&portd 9 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	/* Regulators */
+	lcd_backlight_en {
+		compatible = "regulator-fixed";
+		regulator-name = "lcd_backlight_enable";
+		enable-gpios = <&portc 5 GPIO_ACTIVE_HIGH>;
+		regulator-boot-on;
+	};
+	rpi_power_3v3_en {
+		compatible = "regulator-fixed";
+		regulator-name = "rpi_power_3v3_enable";
+		enable-gpios = <&portc 15 GPIO_ACTIVE_LOW>;
+		regulator-boot-on;
+	};
+	rpi_power_5v_en {
+		compatible = "regulator-fixed";
+		regulator-name = "rpi_power_5v_enable";
+		enable-gpios = <&portc 14 GPIO_ACTIVE_HIGH>;
+		regulator-boot-on;
+	};
+	usb_power_5v_en {
+		compatible = "regulator-fixed";
+		regulator-name = "usb_power_5v_en";
+		enable-gpios = <&porta 27 GPIO_ACTIVE_LOW>;
+	};
+};
+
+&cpu0 {
+	clock-frequency = <120000000>;
```

Comment:
Since the Wio Terminal handles overclocking like a champ, I would consider changing this to the following for folks to be aware they can tweak this:

```suggestion
	clock-frequency = <120000000>; // 120 MHz (default), can be boosted up to 200 MHz
```